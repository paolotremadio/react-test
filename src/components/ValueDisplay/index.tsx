import React, { FunctionComponent } from 'react';
import PropTypes from 'prop-types';

type ValueDisplayProps = {
  value: string
}

export const ValueDisplay: FunctionComponent<ValueDisplayProps> = ({ value }) => (
  <div>{`The value is: ${value}`}</div>
);

ValueDisplay.propTypes = {
  value: PropTypes.string.isRequired,
};
