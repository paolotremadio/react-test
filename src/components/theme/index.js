import { createMuiTheme } from '@material-ui/core';

export const theme = createMuiTheme({
  props: {
    MuiAppBar: {
      elevation: 0,
    },
    MuiButton: {
      disableElevation: true,
    },
    MuiButtonBase: {
      disableRipple: true,
    },
    MuiContainer: {
      maxWidth: 'lg',
    },
    MuiTab: {
      disableFocusRipple: true,
      disableRipple: true,
    },
    MuiTypography: {
      component: 'div',
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 492,
      md: 768,
      lg: 1020,
      xl: 1380,
    },
  },
  palette: {
    primary: {
      light: '#2691de',
      main: '#1b75b6',
    },
    background: {
      default: '#EDEDED',
    },
    text: {
      primary: '#2E353F',
    },
    appBar: {
      background: '#293F54',
      text: '#fff',
    },
  },
  typography: {
    // NOTE: "HelveticaNeue" does not support weights (light, medium); use "Helvetica Neue" instead.
    fontFamily: '"Helvetica Neue", Helvetica, Arial, Roboto, sans-serif',
    //
    // Font scale: Minor third (1.200)
    //
    fontSize: 16,
    h1: {
      fontSize: '2.488rem',
      fontWeight: 500,
    },
    h2: {
      fontSize: '2.074rem',
      fontWeight: 500,
    },
    h3: {
      fontSize: '1.728rem',
      fontWeight: 400,
    },
    h4: {
      fontSize: '1.44rem',
      fontWeight: 400,
    },
    h5: {
      fontSize: '1.2em',
      fontWeight: 500,
    },
    h6: {
      fontSize: '1rem',
      fontWeight: 500,
    },
    button: {
      textTransform: 'initial',
    },
  },
});
