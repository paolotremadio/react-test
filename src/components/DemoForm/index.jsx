import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Formik, Form, Field } from 'formik';
import { TextField } from 'formik-material-ui';
import {
  Box, Divider, Button,
} from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';

import { ValueDisplay } from '@/components/ValueDisplay';

const FieldContainer = withStyles((theme) => ({
  root: {
    paddingBottom: theme.spacing(2.5),
  },
}))(Box);

export const DemoForm = ({ initialValue }) => {
  const [submittedValue, setSubmittedValue] = useState(initialValue);

  return (
    <Formik
      initialValues={{
        value: initialValue,
      }}
      onSubmit={(values, actions) => {
        setSubmittedValue(values.value);
        actions.setSubmitting(false);
      }}
    >
      {({
        submitForm, isSubmitting,
      }) => (
        <Form>
          <FieldContainer>
            <Field
              name="value"
              component={TextField}
              label="Type a value"
              variant="outlined"
              size="medium"
              inputProps={{ 'data-testid': 'input' }}
              fullWidth
            />
          </FieldContainer>
          <FieldContainer>
            <ValueDisplay value={submittedValue} />
          </FieldContainer>
          <Divider />
          <Box paddingTop={2}>
            <Button
              variant="contained"
              color="primary"
              size="large"
              data-testid="button"
              disabled={isSubmitting}
              onClick={submitForm}
            >
              Submit
            </Button>
          </Box>
        </Form>
      )}
    </Formik>
  );
};

DemoForm.propTypes = {
  initialValue: PropTypes.string.isRequired,
};
