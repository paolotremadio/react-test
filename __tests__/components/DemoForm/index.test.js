import {
  render, screen, waitFor, fireEvent,
} from '@testing-library/react';
import { DemoForm } from '@/components/DemoForm';

describe('DemoForm', () => {
  it('Renders the correct initial value', () => {
    render(<DemoForm initialValue="test"/>);

    expect(
      screen.getByText('The value is: test')
    ).toBeInTheDocument();
  });

  it('Updates on submit', async () => {
    const { findByTestId } = render(<DemoForm initialValue="test"/>);

    const input = await findByTestId('input');
    const button = await findByTestId('button');

    fireEvent.change(input, {
      target: {
        value: 'mock',
      },
    });

    fireEvent.click(button);

    await waitFor(() => {
      expect(
        screen.getByText('The value is: mock')
      ).toBeInTheDocument();
    });
  });

  it('Renders correct snapshot', async () => {
    const { container } = render(<DemoForm initialValue="test"/>);

    expect(
      container
    ).toMatchSnapshot();
  });
});
