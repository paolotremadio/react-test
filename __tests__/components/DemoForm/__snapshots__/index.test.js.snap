// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoForm Renders correct snapshot 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="MuiBox-root MuiBox-root-18 Styled(MuiBox)-root-17"
      classes="[object Object]"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink="true"
        >
          Type a value
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="input"
            name="value"
            type="text"
            value="test"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-19 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-21 PrivateNotchedOutline-legendNotched-22"
            >
              <span>
                Type a value
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-23 Styled(MuiBox)-root-17"
      classes="[object Object]"
    >
      <div>
        The value is: test
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiBox-root MuiBox-root-24"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        data-testid="button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Submit
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;
